ably-ios/ARTProtocolMessage+Private.h

```diff
@@ -0,0 +1,23 @@
+//
+//  ARTProtocolMessage+Private.h
+//  ably-ios
+//
+//  Created by Toni Cárdenas on 26/01/2016.
+//  Copyright (c) 2014 Ably. All rights reserved.
+//
+
+ART_ASSUME_NONNULL_BEGIN
+
+@interface ARTProtocolMessage ()
+
+@property (art_nullable, readwrite, strong, nonatomic) NSString *clientId;
```

Comment:
It is implemented kind of backwards. I can see the point, which is to store the ConnectionDetails in the ProtocolMessage itself, and just have the former point to the latter, but I think the other way around (ie. having a ProtocolMessage.connectionKey getter that points to ConnectionDetails.connectionKey if exists) . The API isn't affected, though (this is a private header).

Done at 7bef28a81c3be38f4c3f7dfd84d49e57cf88eac8.